src/main/java/Duke.java

```diff
@@ -1,10 +1,130 @@
+import java.io.*;
+import java.util.Scanner;
+import java.util.*;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    public static void main(String[] args) throws IOException {
+        String line;
+        ArrayList<Task> tasks = new ArrayList<>();
+        Scanner in = new Scanner(System.in);
+        FileInputStream fis=new FileInputStream("ip\\src\\data\\duke.txt");
+        Scanner sc=new Scanner(fis);
+        FileWriter writer = new FileWriter("ip\\src\\data\\duke.txt");
+        final String SPLIT_LINE = "____________________________________________________________";
+        System.out.println(SPLIT_LINE);
+        System.out.println(" Hello! I'm Duke");
+        System.out.println(" What can I do for you?");
+        System.out.println(SPLIT_LINE);
+        if(sc.hasNextLine()){
+            line=sc.nextLine();}
+        else {
+            line=in.nextLine();
+        }
+        while (!line.equals("bye")){
+            try{
+                if(line.equals("list")){
+                    System.out.println(SPLIT_LINE);
+                    for (int i = 0; i < tasks.size(); i++) {
+
+                        System.out.println((i+1)+":"+tasks.get(i).toString());
+                    }
+                    System.out.println(SPLIT_LINE);
```

Comment:
You may want to consider abstracting this out into a method, so as to avoid deep nesting. https://nus-cs2113-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting